### DISCUSSION TITLE
"brew edit" followed by "brew reinstall --build-from-source" doesn't take build changes in consideration

### DISCUSSION THREAD

**Discussion by mathieuatspiria** [Homebrew/discussions#4286] [Casks] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.4
ORIGIN: https://github.com/Homebrew/brew
HEAD: 4d0154fa7926e8a7b06e9b074dd45957e820407c
Last commit: 34 hours ago
Core tap JSON: 28 Feb 21:36 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.86.0 => /usr/bin/curl
macOS: 13.2.1-arm64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2
Rosetta 2: false
```

### Output of `brew doctor`

```shell
==> Downloading https://formulae.brew.sh/api/formula.jws.json
######################################################################## 100.0%
Your system is ready to brew.
```

### Description of issue

I was used  to modify compilation flags for open-scene-graph, but it seems that the process I used no more work. It's possible this issue started after I upgraded homebrew yesterday using "brew upgrade".
Here is what I am doing:

- I do: brew edit open-scene-graph
- I modify build flags for macOS
- I quit VI with :wq.
- Note that if I use brew edit again, my changes are still visible
- Then I do: brew reinstall --build-from-source open-scene-graph
- The process complete successfully like before, but when I use open-scene-graph in my software, it's like if my compile flags parameters were not applied. It was working before.

After some time I decided to uninstall completely brew and reinstall it clean to see if it may fix my issue.
Now when I do the same process I have the same issue, but something new is that now, after I compile and reinstall, if I use "brew edit open-scene-graph" again, then my changes have disappeared and it looks like I never modified it.

I don't know if this issue happens with other packages.

### Relevant casks

open-scene-graph

**Comment by nandahkrishna** [comment, marked as answer]
This is because `brew` fetches the source from GitHub for `--build-from-source` when using the API, which is the new default (and no longer installs based on formulae in the `homebrew-core` repository by default). https://github.com/Homebrew/brew/pull/14807 aims to improve the messaging here.

If you want to use your edited version of the formula, I would **highly** recommend extracting the modified formula to a tap instead. This will allow you to untap `homebrew-core` and take advantage of the API improvements, compared to setting `HOMEBREW_NO_INSTALL_FROM_API`. Some useful docs on taps:

https://docs.brew.sh/Taps
https://docs.brew.sh/Interesting-Taps-and-Forks
https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap
https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/

**Reply by mathieuatspiria** [reply]
Thank you very much for your help. I've been able to make it work thanks to your answer.
